perfmetrics/scripts/continuous_test/gcp_ubuntu/custom_vm_perf_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2024 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# To automate perf tests, this script creates a VM instance based on the
+# flags passed.The command for running the script:
+# python3 custom_vm_perf_test.py --vm_name=<vm_name> --machine_type=<machine-type>
+# image_family=<image_family> --image_project=<image_project> --zone=<zone>
+# --startup_script=<startup_script_filepath>
+
+# For changing the bucket/spreadsheet id, change values in custom_vm_startup_script.sh
+
+import argparse
+import sys
+import subprocess
+
+DEFAULT_VM_NAME = "perf-tests-vm"
+DEFAULT_MACHINE_TYPE = "n2-standard-96"
+DEFAULT_IMAGE_FAMILY = "ubuntu-2004-lts"
+DEFAULT_IMAGE_PROJECT = "ubuntu-os-cloud"
+DEFAULT_ZONE = "us-west1-b"
+DEFAULT_STARTUP_SCRIPT = "./../../custom_vm_startup_script.sh"
+BOOT_DISK_SIZE = "100GiB"
+
+def _parse_arguments(argv):
```

Comment:
do you want to write unit tests for this?

Reply:
added.